app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,58 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  def index
+    @posts = Post.all
+  end
+
+
+  def show
+    @post = Post.friendly.find(params[:id])
```

Comment:
remove this line, since we are getting the post value using filter action

Reply:
removed